src/mat3x4.rs

```diff
@@ -0,0 +1,984 @@
+                - m01 * m13 * m20
+                - m03 * m10 * m21
+                + m03 * m11 * m20,
+        );
+
+        Self::from_rows(x_row, y_row, z_row)
+    }
+
+    /// Multiply with a column vector (`m * v`)
+    #[inline(always)]
+    fn mul_vector4(&self, rhs: Vec4) -> Vec3 {
+        Vec3::new(
+            self.x_row.dot(rhs),
+            self.y_row.dot(rhs),
+            self.z_row.dot(rhs),
+        )
+    }
+
+    /// Multiply with a row vector (`x * m`)
+    /// as if we were a 4x4 matrix (implict `[0, 0, 0, 1]` last row)
+    #[inline(always)]
+    fn mul_row_vec4_from_left(&self, lhs: Vec4) -> Vec4 {
+        Vec4::new(
+            self.x_col_extended_0().dot(lhs),
+            self.y_col_extended_0().dot(lhs),
+            self.z_col_extended_0().dot(lhs),
+            self.w_col_extended_1().dot(lhs),
+        )
+    }
+
+    #[inline]
+    fn look_to_lh(eye: Vec3, dir: Vec3, up: Vec3) -> Self {
+        let f = dir.normalize();
+        let s = up.cross(f).normalize();
+        let u = f.cross(s);
+        Self::from_rows(
+            s.extend(-s.dot(eye)),
+            u.extend(-u.dot(eye)),
+            f.extend(-f.dot(eye)),
+        )
+    }
+
+    /// Creates a left-handed view matrix using a camera position, an up direction, and a focal
+    /// point.
+    #[inline]
+    pub fn look_at_lh(eye: Vec3, center: Vec3, up: Vec3) -> Self {
+        glam_assert!(up.is_normalized());
+        Self::look_to_lh(eye, center - eye, up)
+    }
+
+    /// Creates a right-handed view matrix using a camera position, an up direction, and a focal
+    /// point.
+    #[inline]
+    pub fn look_at_rh(eye: Vec3, center: Vec3, up: Vec3) -> Self {
+        glam_assert!(up.is_normalized());
+        Self::look_to_lh(eye, eye - center, up)
+    }
+
+    /// Transforms the given 3D vector as a point.
+    ///
+    /// This is the equivalent of multiplying the 3D vector as a 4D vector where `w` is
+    /// `1.0`.
+    #[inline]
+    pub fn transform_point3(&self, point: Vec3) -> Vec3 {
+        self.mul_vector4(point.extend(1.0))
+    }
+
+    /// Transforms the give 3D vector as a direction.
+    ///
+    /// This is the equivalent of multiplying the 3D vector as a 4D vector where `w` is
+    /// `0.0`.
+    #[inline]
+    #[cfg(any(
+        target_arch = "spirv",
+        all(target_feature = "sse2", not(feature = "scalar-math"))
+    ))]
+    pub fn transform_vector3(&self, vec: Vec3) -> Vec3 {
+        self.mul_vector4(vec.extend(0.0))
+    }
+
+    /// Transforms the give 3D vector as a direction.
+    ///
+    /// This is the equivalent of multiplying the 3D vector as a 4D vector where `w` is
+    /// `0.0`.
+    #[inline]
+    #[cfg(not(any(
+        target_arch = "spirv",
+        all(target_feature = "sse2", not(feature = "scalar-math"))
+    )))]
+    pub fn transform_vector3(&self, vec: Vec3) -> Vec3 {
+        Vec3::new(
+            Vec3::from(self.x_row).dot(vec),
```

Comment:
If you use `Vec3A` here it will probably perform better.

Reply:
No - this is the scalar path, and Vec3A is (slightly) slower (3.1 ns vs 2.9 ns)

Reply:
Ah my bad, I missed the outer `not`